research/fed-sm/job_configs/fedsm_prostate/app/custom/persistors/pt_file_fedsm_model_persistor.py

```diff
@@ -0,0 +1,218 @@
+                data = torch.load(src_file_name, map_location=device)
+                # checkpoint may contain a dict "model_set_fedsm" of models indexed with model ids
+                # 'optimizer', 'lr_scheduler', etc.
+            except:
+                self.log_exception(fl_ctx, "error loading checkpoint from {}".format(src_file_name))
+                self.system_panic(reason="cannot load model checkpoint", fl_ctx=fl_ctx)
+                return None
+        else:
+            # if no pretrained model provided, use the generated network weights from APP config_3
+            # note that, if set "determinism" in the config_3, the init model weights will always be the same
+            try:
+                data["model_set_fedsm"]["select_weights"] = (
+                    self.model_set_fedsm["select_weights"].state_dict()
+                    if self.model_set_fedsm["select_weights"] is not None
+                    else OrderedDict()
+                )
+                data["model_set_fedsm"]["select_exp_avg"] = OrderedDict()
+                data["model_set_fedsm"]["select_exp_avg_sq"] = OrderedDict()
+                data["model_set_fedsm"]["global_weights"] = (
+                    self.model_set_fedsm["global_weights"].state_dict()
+                    if self.model_set_fedsm["global_weights"] is not None
+                    else OrderedDict()
+                )
+                for id in self.client_ids:
+                    data["model_set_fedsm"][id] = (
+                        self.model_set_fedsm[id].state_dict() if self.model_set_fedsm[id] is not None else OrderedDict()
+                    )
+            except:
+                self.log_exception(fl_ctx, "error getting state_dict from model object")
+                self.system_panic(reason="cannot create state_dict from model object", fl_ctx=fl_ctx)
+                return None
+
+        if self.model and self.model_selector:
+            self.default_train_conf = {
+                "train": {
+                    "model": type(self.model).__name__,
+                    "model_selector": type(self.model_selector).__name__,
+                }
+            }
+
+        self.persistence_manager = PTModelPersistenceFormatManagerFedSM(
+            data, default_train_conf=self.default_train_conf
+        )
+        return self.persistence_manager.to_model_learnable(self.exclude_vars)
+
+    def handle_event(self, event: str, fl_ctx: FLContext):
+        if event == EventType.START_RUN:
+            self._initialize(fl_ctx)
+        elif event == "fedsm_best_global_model_available":
+            # save the current model as the best model!
+            self.save_model_file_global(self.best_ckpt_save_path_global)
+        elif event == "fedsm_best_select_model_available":
+            # save the current model as the best model!
+            self.save_model_file_select(self.best_ckpt_save_path_select)
+
+    def save_model_file_global(self, save_path: str):
```

Comment:
`save_model_file_global` and `save_model_file_global` have the same implementation. Is this intended?

Reply:
fixed